akka-stream/src/main/boilerplate/akka/stream/FanOutShape.scala.template

```diff
@@ -53,14 +53,20 @@ object UniformFanOutShape {
 }
 
 class UniformFanOutShape[-I, +O](n: Int, _init: FanOutShape.Init[I @uncheckedVariance]) extends FanOutShape[I](_init) {
+
+  for (i <- 0 until n) newOutlet[O](s"out$i")
+
   def this(n: Int) = this(n, FanOutShape.Name[I]("UniformFanOut"))
   def this(n: Int, name: String) = this(n, FanOutShape.Name[I](name))
   def this(inlet: Inlet[I], outlets: Array[Outlet[O]]) = this(outlets.size, FanOutShape.Ports(inlet, outlets.toList))
   override protected def construct(init: FanOutShape.Init[I @uncheckedVariance]): FanOutShape[I] = new UniformFanOutShape(n, init)
   override def deepCopy(): UniformFanOutShape[I, O] = super.deepCopy().asInstanceOf[UniformFanOutShape[I, O]]
-  
-  val outArray: Array[Outlet[O @uncheckedVariance]] = Array.tabulate(n)(i => newOutlet[O](s"out$i"))
-  def out(n: Int): Outlet[O @uncheckedVariance] = outArray(n)
+
+  final override def outlets: immutable.Seq[Outlet[O @uncheckedVariance]] = super.outlets.asInstanceOf[immutable.Seq[Outlet[O]]]
+
+  @deprecated("use `outlets` or `out(id)` instead", "2.5.4")
+  lazy val outArray: Array[Outlet[O @uncheckedVariance]] = outlets.toArray
```

Comment:
Since this one, unlike the Seq may have been used by Java people, we should annotate it as Java's Deprecated as well. See here: https://github.com/akka/akka/pull/21913/files#diff-ac833d777f0270e843a0ed2af8aae769R23 This is to make sure it also shows up as deprecated for Java users without any scala plugins etc